main/src/main/java/tech/pegasys/poc/witnesscodeanalysis/WitnessCodeAnalysis.java

```diff
@@ -14,214 +14,173 @@
  */
 package tech.pegasys.poc.witnesscodeanalysis;
 
-import com.google.gson.Gson;
-import com.google.gson.GsonBuilder;
 import org.apache.logging.log4j.Logger;
 import org.apache.tuweni.bytes.Bytes;
-import tech.pegasys.poc.witnesscodeanalysis.common.ChunkData;
 import tech.pegasys.poc.witnesscodeanalysis.common.ContractData;
-import tech.pegasys.poc.witnesscodeanalysis.fixed.FixedSizeAnalysis;
-import tech.pegasys.poc.witnesscodeanalysis.functionid.FunctionIdAllLeaves;
-import tech.pegasys.poc.witnesscodeanalysis.functionid.FunctionIdProcess;
-import tech.pegasys.poc.witnesscodeanalysis.jumpdest.JumpDestAnalysis;
-
-import java.io.BufferedReader;
-import java.io.FileWriter;
-import java.io.PrintWriter;
-import java.io.StringWriter;
-import java.io.Writer;
-import java.nio.charset.StandardCharsets;
-import java.nio.file.Files;
-import java.nio.file.Path;
-import java.nio.file.Paths;
-import java.util.ArrayList;
-
-import static java.lang.System.exit;
-import static org.apache.logging.log4j.LogManager.getLogger;
+import tech.pegasys.poc.witnesscodeanalysis.processing.FixedSizeProcessing;
+import tech.pegasys.poc.witnesscodeanalysis.processing.FunctionIdProcessing;
+import tech.pegasys.poc.witnesscodeanalysis.processing.JumpDestProcessing;
+import tech.pegasys.poc.witnesscodeanalysis.processing.SimpleProcessing;
+import tech.pegasys.poc.witnesscodeanalysis.processing.StrictFixedSizeProcessing;
 
-public class WitnessCodeAnalysis extends CodeAnalysisBase {
-  private static final Logger LOG = getLogger();
+import java.io.IOException;
 
-  public static final String DEFAULT_FILE_IN =  "contract_data.json";
-  public static final String DEFAULT_JUMPDEST_FILE_OUT =  "analysis_jumpdest.json";
-  public static final String DEFAULT_FIXED_FILE_OUT =  "analysis_fixed.json";
-  public static final String DEFAULT_STRICTFIXED_FILE_OUT =  "analysis_strictfixed.json";
+import static org.apache.logging.log4j.LogManager.getLogger;
 
-  private static final int THRESHOLD = 128;
+public class WitnessCodeAnalysis {
+  private static final Logger LOG = getLogger();
 
-  public WitnessCodeAnalysis(Bytes code) {
-    super(code);
+  public static boolean SIMPLE = true;
+  public static boolean JUMPDEST = true;
+  public static boolean FIXEDSIZE = true;
+  public static boolean STRICTFIXEDSIZE = true;
+  public static boolean FUNCTIONID = true;
+
+  private MainNetContractDataSet dataSet;
+  private SimpleProcessing simpleProcessing;
+  private JumpDestProcessing jumpDestProcessing;
+  private FixedSizeProcessing fixedSizeProcessing;
+  private StrictFixedSizeProcessing strictFixedSizeProcessing;
+  private FunctionIdProcessing functionIdProcessing;
+
+
+  public WitnessCodeAnalysis() throws IOException {
+    this.dataSet = new MainNetContractDataSet();
+    this.simpleProcessing = new SimpleProcessing(true);
```

Comment:
Do you intend to use constants SIMPLE, etc here in place of 'true' for constructor?